zebra_app/app.R

```diff
@@ -48,35 +48,98 @@ ui <- page_sidebar(
+            nav_panel("Mean Activity", plotOutput("plot4")),
+            nav_panel("Percent Change", plotOutput("plot5")),
+            nav_panel("Activity by Period", plotOutput("plot6")),
+            nav_panel("Percent Change by Period", plotOutput("plot7")),
+            nav_panel("Statistical Summary", DT::dataTableOutput("stats_table")),
+            nav_panel("Plate Map", plotlyOutput("plate_map")),
+            nav_panel("Cleaned Data", DT::dataTableOutput("cleaned_table"))
           )
         )
       )
     )
   ),
   
   theme = bs_theme(
-    bg = "white",
+    bg = "#FAFAF8",
     fg = "black",
-    primary = "#E69F00",
-    secondary = "#0072B2",
-    success = "#009E73",
-    base_font = font_google("Inter")
+    primary = "#429E99",
+    secondary = "#B2DFDB",
+    success = "#00796B",
+    base_font = font_google("Inter"),
+    bootswatch = "flatly"
   )
 )
-  
+
 server <- function(input, output, session) {
   
+  metadata <- reactive({
+    req(input$metadata_file)
+    readxl::read_excel(input$metadata_file$datapath)
+  })
+  
+  zebrabox_data <- reactive({
+    req(input$data_file)
+    raw_data <- read.delim(input$data_file$datapath, sep = "\t", header = TRUE)
+    
+    raw_data %>% 
+      mutate(time_min = start / 60, 
+             # experiment_time_of_day = '',
+             period2 = ifelse(nchar(time_min) == 1, 1,
+                              as.integer(str_extract(time_min, '^[0-9]')) + 1),
+             plate = str_extract(location, 'Loc[AB]'),
+             light = ifelse(period2 %% 2 != 0 & period2 != 1, 'dark', 'light')) %>%
+      separate(aname, into = c('well', 'name'), sep = '_') %>%
+      left_join(metadata(), by = join_by(well, plate == box_used)) %>%
+      select(plate, treatment, well, time_min, light, period, 
+             datatype, activity = actinteg, everything()) %>%
+      filter(!is.na(treatment), timebinid == 1) 
+  })
+  
+  output$cleaned_table <- DT::renderDataTable({
+    req(zebrabox_data())
+    DT::datatable(zebrabox_data(),
+                  extensions = 'Buttons',
+                  options = list(dom = 'Blfrtip',
+                                 buttons = c('copy', 'csv', 'excel'),
+                                 lengthMenu = list(c(10, 25, 50, -1),
+                                                   c(10, 25, 50, "All"))),
+                  caption = htmltools::tags$caption(
+                    style = 'caption-side: top; text-align: left; color:black; font-size:200%;',
+                    'Zebrabox'))
+  })
+  
   output$download_template <- downloadHandler(
-    filename = function() "metadata_template.csv",
-    content = function(file) {
-      write.csv(data.frame(SampleID = "", Condition = ""), file, row.names = FALSE)
+    filename = function() { 
+      "0000-00-00_zebrabox_experiment_template.xlsx"
+    },
+    content = function(file) {file.copy("C:/Users/layla/OneDrive/Documents/GitHub/zebrabox/0000-00-00_zebrabox_experiment_template.xlsx", file)
```

Comment:
This needs to be the relative path to the template in the Git repository. `C:/Users/layla/OneDrive/etc..` is the absolute path on your computer so it won't work for anyone else